tests/base.py

```diff
@@ -625,6 +649,9 @@ def assertPKsEqual(self, stream, expected_records, sync_records):
             # Verify sync pks have all expected records pks in it
             self.assertTrue(sync_pks_set.issuperset(expected_pks_set))
 
+            if assert_pk_count_same:
```

Comment:
@kspeer825 this is the flag I added which make the test work as expected. Refunds and payments both ignore the cursor and return all pages if an invalid cursor / cursor that doesn't match the previous query is sent.